lib/AST/ASTPrinter.cpp

```diff
@@ -3455,6 +3460,18 @@ void Pattern::print(llvm::raw_ostream &OS, const PrintOptions &Options) const {
 //  Type Printing
 //===----------------------------------------------------------------------===//
 
+template <typename ExtInfo>
+void printCType(ASTContext &Ctx, ASTPrinter &Printer, ExtInfo &info) {
+  auto *cml = Ctx.getClangModuleLoader();
+  SmallString<64> buf1;
+  llvm::raw_svector_ostream os1(buf1);
+  info.getUncommonInfo().getValue().printClangFunctionType(cml, os1);
+  SmallString<64> buf2;
+  llvm::raw_svector_ostream os2(buf2);
+  os2 << QuotedString(os1.str());
+  Printer << ", cType: " << os2.str();
```

Comment:
You don't need the second buffer here.

Reply:
There is not overload for `<<` that takes a `QuotedString` -- if I add that, that would only shift buffer from one spot to another, so I'm not sure if that is the solution you were looking for.

Reply:
I added that missing overload and then used that here.

Reply:
What?  Oh, no overload for `ASTPrinter`.  Sure.